### DISCUSSION TITLE
No longer open-source?

### DISCUSSION THREAD

**Discussion by ErikSwan** [jgraph/drawio#4819] [General]
I could have sworn the draw.io source code used to be fully open-source, but I've come back recently to report a couple of bugs and I noticed that a lot of the source code in the GitHub repository is now minified, and the README now states (emphasis mine):

> The minified code authored by us in this repo is licensed under an Apache v2 license, but the sources to build those files are not in this repo. **This is not an open source project.**

This is a big bummer, because it makes it pretty much impossible for me to go and fix annoying bugs I run into myself and then contribute those fixes back.

When did this change occur? Is there an announcement or discussion thread or blog post where this change to the license was announced? What was the reasoning behind the change?

I can't find anything in either the GitHub Discussions, or on the [Draw.io Blog](https://www.drawio.com/blog), so I'm left feeling like this was a stealth change.

**Comment by dvaerum** [comment]
They must have deleted the original post. But apparently somebody named “Dave Kinne” is to blame.

I found the link to the original post thanks to the NIX people ❤️
https://github.com/NixOS/nixpkgs/commit/da2060c0dee86a677096aab104c029a578bd12ea

You can read more about it at archive.org
https://web.archive.org/web/20241003003512/https://github.com/jgraph/drawio/discussions/4623

But since they deleted the original post, I assume they are also going to delete this one too — now.
Which I think is bullshit. If you are going to change the license or remove the source files, stand by your decision instead of trying to hide it.
It sucks that Draw.io decided to go this way, but I don't think that they have done something wrong. I just hope they start going open source again.

**Comment by davidjgraph** [comment]
Why exactly it's a "stealth" change or "hidden" I find "unusual" given you quoted the README that clearly says near the top exactly the thing that you're suggesting is being hidden. The README is the most visible place to indicate the license state.

In terms of being able to submit fixes, draw.io was closed for contribution the entire time it was open source, this how we hold full copyright over the codebase and are able to change the license.

If this thread remains constructive it can stay, but if it just descends into complaining until the change is reverted, it will be pruned. The change is not going to be reverted, if people post revert requests thinly veiled as information posts, I'll delete them.

**Reply by dvaerum** [reply]
> Why exactly it's a "stealth" change or "hidden" I find "unusual" given you quoted the README that clearly says near the top exactly the thing that you're suggesting is being hidden. The README is the most visible place to indicate the license state.

Because a previous discussion was removed that explained what happened, and the README also does not say anything about why

> The source code authored by us in this repo is licensed under a modified Apache v2 license. This project is not an open source 
project as a result

became

> The minified code authored by us in this repo is licensed under an Apache v2 license, but the sources to build those files are not in this repo

But as I said, “I don't think that they have done something wrong”. I just think that it sucks that the project went that way because of some random person on the Internet, but “Dave Kinne” what probably not the first.

Anyway @davidjgraph, I don't think there is more to discuss, so if possible (on Github) you can just lock the discussion, I am not here to change your mind. I just wanted to answer @ErikSwan question because I had the same question and found the answer.

**Comment by madhairsilence** [comment]
I understand the source code is no longer provided in this repo and I get only the minified. Minified or expanded , its a still a JS code not a binary. And the code is still under Apache License ( modified ?? ).

So the question is here, can I still use the code ? and if needed unminify ? Does it violate the license ?
